### PR TITLE
Fixed and updated rke upgrade tests 

### PR DESCRIPTION
**Pull request by soumyalj**
@sangeethah : Please review.

### CODE REVIEW

File: lib/node.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def ready_node(self):
 
     def docker_ps(self, all=False):
         result = self.execute_command(
-            'docker ps --format "{{.Names}}\t{{.Image}}"')
+            'docker ps -a --format "{{.Names}}\t{{.Image}}"')
```
Comment (sangeethah) [inline lib/node.py]: @soumyalj Why do we want to look for docker process is stopped stated as well ? This method could be used in places to check strictly for docker process that are in running state.

File: lib/node.py [hunk 2]
```diff
@@ -126,7 +126,9 @@ def docker_inspect(self, container_name, output_format=None):
     def docker_exec(self, container_name, cmd):
         command = 'docker exec {0} {1}'.format(container_name, cmd)
         result = self.execute_command(command)
-        if result[1] != '':
```
Comment (sangeethah) [inline lib/node.py]: Why do we not want to treat warning shown in stderror as an exception in this case ?

File: tests/rke/test_upgrade.py [hunk 3]
```diff
@@ -115,37 +154,3 @@ def test_upgrade_3(test_name, cloud_provider, rke_client, kubectl):
     validate_rke_cluster(
         rke_client, kubectl, after_upgrade_nodes, 'afterupgrade')
     delete_nodes(cloud_provider, all_nodes)
-
```
Comment (sangeethah) [inline tests/rke/test_upgrade.py]: Can we skip this test case for now and revisit this workflow later.

Comment (soumyalj) [reply tests/rke/test_upgrade.py]: Why should it be skipped ?

File: lib/node.py (hunk 2 above)
Comment (soumyalj) [reply lib/node.py]: This is to handle the case where etcdctl issues warning when the cluster health is healthy for some k8s versions. If we don't handle this case, then exception will be raised even when the cluster is healthy for some k8s versions.

File: tests/rke/test_upgrade.py (hunk 3 above)
Comment (soumyalj) [reply tests/rke/test_upgrade.py]: Skip code added

File: lib/node.py (hunk 1 above)
Comment (soumyalj) [reply lib/node.py]: This is because service-sidekick container shows up only in `docker ps -a`.
Modified code to handle this case.

**Comment by sangeethah** [comment]
LGTM

**Pull request merged**